### PR TITLE
AGL US 29-30

### PR DESCRIPTION
**Pull request by alex-latham**
## All Submissions:

* [X] Merge the target branch into the PR branch. Fix any conflicts that might appear.
* [X] Have you checked to ensure that other pull request are approved for merge [Pull Requests](../../../pulls)?
* [X] What stories does the PR include
  - 27-30
### New Feature Submissions:

* [X] Have you added an explanation of what your changes do?
* [X] Have you lint your code locally prior to submission?

### Testing:

* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully ran tests with your changes locally?
* [X] All new and existing tests passed.

**Test coverage** = 100% models and features
--------------------------------------------------------------------------------
### Change Log
All changes from PR20 plus:
- New profile order show page
- Profile order cancellation
- Status added to item_orders table
- Order#cancel added
- Implement resources in profile namespace
- Relocate require_current_user to ApplicationController (used by two controllers)

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
     item_orders.sum('price * quantity')
   end
+
+  def total_quantity
+    item_orders.sum(:quantity)
+  end
+
+  def cancel
+    item_orders.where(status: "fulfilled").each do |item_order|
+      Item.find(item_order.item_id).increment(:inventory, by = item_order.quantity).save!
```
Comment (tylertomlinson) [inline app/models/order.rb:18]: Solid logic

File: app/models/order.rb [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+    end
+    item_orders.update(status: "unfulfilled")
```
Comment (tylertomlinson) [inline app/models/order.rb:20]: This will help

File: app/views/profile/orders/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<p>Date updated: <%= @order.updated_at %></p>
+<p>Status: <%= @order.status %></p>
+<% @order.item_orders.each do |item_order| %>
+  <article id="item_order-<%= item_order.item.id %>">
+    <p>Name: <%= item_order.item.name %></p>
+    <p>Description: <%= item_order.item.description %></p>
+    <%= image_tag item_order.item.image %>
+    <p>Quantity: <%= item_order.quantity %></p>
+    <p>Price: <%= item_order.item.price %></p>
+    <p>Subtotal: <%= item_order.subtotal %></p>
```
Comment (tylertomlinson) [inline app/views/profile/orders/show.html.erb:13]: can we use the `number_to_currrency` here

File: app/views/profile/orders/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  </article>
+<% end %>
+<p>Total quantity: <%= @order.total_quantity %></p>
+<p>Grand total: <%= @order.grandtotal %></p>
```
Comment (tylertomlinson) [inline app/views/profile/orders/show.html.erb]: can we use the number_to_currrency here

File: db/migrate/20200411005517_add_status_to_orders.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToOrders < ActiveRecord::Migration[5.1]
+  def change
+    add_column :orders, :status, :string, default: 'pending'
```
Comment (tylertomlinson) [inline db/migrate/20200411005517_add_status_to_orders.rb:3]: wondering if its better to have the db use integers here 
so the default would be 0

File: db/migrate/20200411214305_add_status_to_item_orders.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToItemOrders < ActiveRecord::Migration[5.1]
+  def change
+    add_column :item_orders, :status, :string, default: 'unfulfilled'
```
Comment (tylertomlinson) [inline db/migrate/20200411214305_add_status_to_item_orders.rb:3]: same here

Comment (tylertomlinson) [reply db/migrate/20200411214305_add_status_to_item_orders.rb:3]: I don't think it matters but I'll look into it

File: spec/features/profile/orders/index_spec.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
```
Comment (tylertomlinson) [inline spec/features/profile/orders/index_spec.rb:1]: solid testing

**Review by tylertomlinson — APPROVED** [review]
Should change your name to Alex hits every edge case on testing Latham. Good work!

**Review by Mariana-21 — APPROVED** [review]

**Pull request merged**